PokeCharts.IntegrationTests/Client/PokeAPIClientTests.cs

```diff
@@ -0,0 +1,106 @@
+using NUnit.Framework;
+using Newtonsoft.Json.Linq;
+using System.Linq;
+
+namespace Client.nUnitTests
+{
+
+    //  This class tests the sending and receiving of requests to the PokeAPI
+    public class PokeAPIClientTest
+    {
+
+        PokeAPIClient _client = new PokeAPIClient();
+        
+        // To prove that we really received data
+        protected int? GetPokemonId(JObject jsonOutput){
+            var pokemons = from pokemon in jsonOutput?["data"]?["pokemon_v2_pokemon"] select pokemon;
+            JToken? firstPokemon = pokemons.FirstOrDefault();
+            return firstPokemon?["id"]?.Value<int>();
+        }
+
+        [Test]
+        public void Execute_emptyQuery_throwException()
+        {
+            // given
+            string query = "";
+
+            // when
+            var exception = Assert.ThrowsAsync<Exception>(() => _client.Execute(query));
+
+            // Then
+            // The returned message should contain: not a valid graphql query
+            Assert.IsNotNull(exception?.Message);
+        }
+
+        [Test]
+        public void Execute_notAQuery_throwException()
+        {
+            // given
+            string query = "Invalid query";
+
+            // when
+            var exception = Assert.ThrowsAsync<Exception>(() => _client.Execute(query));
+
+            // Then
+            // The returned message is should contain: not a valid graphql query
+            Assert.IsNotNull(exception?.Message);
+        }
+
+                [Test]
+        public void Execute_incorrectQuery_throwException()
+        {
+            // given            /
+            string query = @"query {
+                pokemon_v3_pokemon(limit: 3, order_by: {id: asc}) {
```

Comment:
If the "pokemon_v3_pokemon" is the only thing that make the query incorrect, maybe it's risky (PokeApi can probably rename the next version like this) and it will not throw an exception